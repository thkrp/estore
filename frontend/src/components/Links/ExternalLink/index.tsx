import React, { FC } from 'react';
import { sanitizeUrl } from '@braintree/sanitize-url';

type Props = {
    href: string;
    title: string;
    className?: string;
};

const ExternalLink: FC<Props> = ({ href, title, className }) => {
    return (
        <a href={sanitizeUrl(href)} target="_blank" rel="noreferrer noopener" className={className}>
            {title}
        </a>
    );
};

export default ExternalLink;
