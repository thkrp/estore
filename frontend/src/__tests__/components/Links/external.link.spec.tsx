/* eslint-disable no-restricted-imports*/
import React from 'react';
import '@testing-library/jest-dom';
import { cleanup, render, screen } from '@testing-library/react';
import * as sanitizer from '@braintree/sanitize-url';
import ExternalLink from '../../../components/Links/ExternalLink';

describe('ExternalLink', () => {
    afterEach(cleanup);

    it('should contain target="_blank" & rel="noreferrer noopener" attributes', () => {
        render(<ExternalLink href="google.com" title="google" className="external-link" />);
        const element = screen.getByRole('link');
        expect(element).toHaveAttribute('target', '_blank');
        expect(element).toHaveAttribute('rel', 'noreferrer noopener');
    });

    it('should be able to pass a class', () => {
        render(<ExternalLink href="google.com" title="google" className="external-link" />);
        const element = screen.getByRole('link');
        const classes = element.className.split(' ');
        expect(classes.includes('external-link')).toBe(true);
    });

    it('sanitizeUrl has been called', () => {
        jest.spyOn(sanitizer, 'sanitizeUrl');
        const { container } = render(<ExternalLink href="google.com" title="google" />);
        expect(sanitizer.sanitizeUrl).toHaveBeenCalled();
        expect(container).toMatchSnapshot();
    });

    it('should be able to pass a title', () => {
        render(<ExternalLink href="google.com" title="google" className="external-link" />);
        expect(screen.getByText(/google/i)).toBeInTheDocument();
    });
});
