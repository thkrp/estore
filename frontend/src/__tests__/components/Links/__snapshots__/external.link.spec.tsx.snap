// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink sanitizeUrl has been called 1`] = `
<div>
  <a
    href="google.com"
    rel="noreferrer noopener"
    target="_blank"
  >
    google
  </a>
</div>
`;
